### ISSUE TITLE
[Encoding] Non-one-liner custom copy methods cause them to be generated twice

### ISSUE THREAD

**Issue by Alex1304** [immutables/immutables#1305]
In an encoding object, make a custom copy method that isn't a one-liner:

```java
@Encoding
public class CustomValueEncoding {

    @Encoding.Impl
    private CustomValue value;

    // Trivial example that can be easily replaced with a one liner
    // but you can imagine something more complex
    @Encoding.Copy
    public CustomValue with(SomeType v) {
        Objects.requireNonNull(v);
        return new CustomValue(v);
    }
}
```

It will result in an error along these lines:
```
error: withXxx(SomeType) is already defined in ImmutableXxx
```
If you open the generated Immutable object, you realize the method was generated twice, once as a copy method and once as a regular member method (as if it didn't have the `@Encoding.Copy`). If you change the code above to a one-liner (`return new CustomValue(Objects.requireNonNull(v));`), the error will disappear.

Reproducible on both 2.8.8 and 2.9.0-beta2

**Comment by elucash** [comment]
hm, interesting, thank you for reporting this

**Comment by elucash** [comment]
seems to be fixed in master, next beta — in a couple of days

**Comment by elucash** [comment]
took a bit longer  :(  to get to this and add some more fixed, but it's released now in `2.9.0-beta3`
